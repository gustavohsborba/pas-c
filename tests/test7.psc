var
    s, p is int;
    var is int;

begin
    s := 0;

    do
        out({PreÃ§o? });
        in(p);
        s := s+p;
    while p!=0;

    out({Total writeln a pagar: R$});
    out(s);
end.