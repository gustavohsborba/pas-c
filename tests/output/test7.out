Opening file tests/test7.psc...
Token found: (TOK_VAR)
Token found: (TOK_ID, s)
Token found: (TOK_COMMA)
Token found: (TOK_ID, p)
Token found: (TOK_IS)
Token found: (TOK_INT)
Token found: (TOK_SEMICOLON)
Token found: (TOK_VAR)
Token found: (TOK_IS)
Token found: (TOK_INT)
Token found: (TOK_SEMICOLON)
Token found: (TOK_BEGIN)
Token found: (TOK_ID, s)
Token found: (TOK_ASSIGN)
Token found: (TOK_CONST_INT, 0)
Token found: (TOK_SEMICOLON)
Token found: (TOK_DO)
Token found: (TOK_OUT)
Token found: (TOK_PAR_OPEN)
Token found: (TOK_CONST_STR, Valor a ser somado: )
Token found: (TOK_PAR_CLOSE)
Token found: (TOK_SEMICOLON)
Token found: (TOK_IN)
Token found: (TOK_PAR_OPEN)
Token found: (TOK_ID, p)
Token found: (TOK_PAR_CLOSE)
Token found: (TOK_SEMICOLON)
Token found: (TOK_ID, s)
Token found: (TOK_ASSIGN)
Token found: (TOK_ID, s)
Token found: (TOK_ADD)
Token found: (TOK_ID, p)
Token found: (TOK_SEMICOLON)
Token found: (TOK_WHILE)
Token found: (TOK_ID, p)
ERROR in line 12, column 12: Token ! is not an operator
Token found: (TOK_EQUALS)
Token found: (TOK_CONST_INT, 0)
Token found: (TOK_SEMICOLON)
Token found: (TOK_OUT)
Token found: (TOK_PAR_OPEN)
Token found: (TOK_CONST_STR, Total a pagar: R$)
Token found: (TOK_PAR_CLOSE)
Token found: (TOK_SEMICOLON)
Token found: (TOK_OUT)
Token found: (TOK_PAR_OPEN)
Token found: (TOK_ID, s)
Token found: (TOK_PAR_CLOSE)
Token found: (TOK_SEMICOLON)
Token found: (TOK_END)
ERROR in line 16, column 5: Token . is not an operator
Token found: (TOK_EOF)
